inference-engine/src/inference_engine/include/openvino/runtime/gpu/dx.hpp

```diff
@@ -0,0 +1,192 @@
+        // clang-format off
+        ParamMap contextParams = {
+            {GPU_PARAM_KEY(CONTEXT_TYPE), GPU_PARAM_VALUE(VA_SHARED)},
+            {GPU_PARAM_KEY(VA_DEVICE), static_cast<gpu_handle_param>(device)}
+        };
+        // clang-format on
+        return std::dynamic_pointer_cast<D3DContext>(core.CreateContext(deviceName, contextParams));
+    }
+
+    /**
+     * @brief This function is used to obtain a NV12 tensor from NV12 DXGI video decoder output.
+     * The resulting tensor contains two remote tensors for Y and UV planes of the surface.
+     * @param height Height of Y plane
+     * @param width Widht of Y plane
+     * @param nv12_surf A ID3D11Texture2D instance to create NV12 blob from
+     * @return NV12 remote tensor
+     */
+    RemoteTensor create_tensor_nv12(const size_t height, const size_t width, const ID3D11Texture2D* nv12_surf) {
+        ParamMap blobParams = {{GPU_PARAM_KEY(SHARED_MEM_TYPE), GPU_PARAM_VALUE(VA_SURFACE)},
+                               {GPU_PARAM_KEY(DEV_OBJECT_HANDLE), static_cast<gpu_handle_param>(nv12_surf)},
+                               {GPU_PARAM_KEY(VA_PLANE), uint32_t(0)}};
+        auto y_tensor = create_tensor(element::u8, {1, 1, height, width}, blobParams);
+        blobParams[GPU_PARAM_KEY(MEM_HANDLE)] = static_cast<gpu_handle_param>(nv12_surf);
+        blobParams[GPU_PARAM_KEY(VA_PLANE)] = uint32_t(1);
+        auto uv_tensor = create_tensor(element::u8, {1, 2, height / 2, width / 2}, blobParams);
+        return Tensor::create_nv12(y_tensor, uv_tensor);
+    }
+
+    /**
+     * @brief This function is used to obtain remote blob object from ID3D11Buffer
+     * @param type Tensor element type
+     * @param shape Tensor shape
+     * @param buffer A pointer to ID3D11Buffer instance to create remote blob based on
+     * @return A remote blob instance
+     */
+    D3DBufferTensor create_tensor(const element::Type type, const Shape& shape, const ID3D11Buffer* buffer) {
+        ParamMap params = {{GPU_PARAM_KEY(SHARED_MEM_TYPE), GPU_PARAM_VALUE(DX_BUFFER)},
+                           {GPU_PARAM_KEY(DEV_OBJECT_HANDLE), static_cast<gpu_handle_param>(buffer)}};
+        create_tensor(type, shape, params);
+    }
+
+    /**
+     * @brief This function is used to obtain remote tensor object from ID3D11Texture2D
+     * @param type Tensor element type
+     * @param shape Tensor shape
+     * @param surface Pointer to ID3D11Texture2D interface of the objects that owns NV12 texture
+     * @param plane ID of the plane to be shared (0 or 1)
+     * @return D3DSurface2DTensor tensor
+     * @note The underlying ID3D11Texture2D can also be a plane of output surface of DXGI video decoder
+     */
+    D3DSurface2DTensor create_tensor(const element::Type type,
+                                     const Shape& shape,
+                                     ID3D11Texture2D* surface,
+                                     uint32_t plane = 0) {
```

Comment:
looks like this function is generic one and can totally replace NV12 `create_tensor_nv12`  with `ID3D11Texture2D`

Reply:
@vladimir-paramuzov What about `GPU_PARAM_KEY(MEM_HANDLE)` parameter. It is not set by that function but used in `create_tensor_nv12`